tests/unit_tests/analysis/test_es_update.py

```diff
@@ -726,19 +749,7 @@ def test_temporary_parameter_storage_with_inactive_fields(
         ensemble, param_ensemble_array, param_group, realization_list
     )
     for iens in range(prior_ensemble.ensemble_size):
-        ds = xr.open_dataset(
-            ensemble._path / f"realization-{iens}" / f"{param_group}.nc", engine="scipy"
+        ds = ensemble.load_parameters(param_group, iens)
+        np.testing.assert_array_equal(
+            ds["values"].values[0], fields[iens]["values"].values[0]
         )
-        np.testing.assert_array_equal(ds["values"].values[0], fields[iens]["values"])
-
-
-def test_that_observations_keep_sorting(snake_oil_case_storage, snake_oil_storage):
```

Comment:
We should have some test checking that the order is the same at least (changing the order is ok if we have to, but would like to know if it changes in the future).

Reply:
Will set up a test. I think the case we want to always be true is: The matrix S has its rows sorted by `[realization, name, ...primary_key]` where `primary_key` is whatever the response specifies. This also only applies for the combined datasets, if we sort after a combine we'll always know it is sorted.